pkg/registry/core/rest/storage_core.go

```diff
@@ -325,13 +283,32 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(apiResourceConfigSource
 	}
 
 	if resource := "replicationcontrollers"; apiResourceConfigSource.ResourceEnabled(corev1.SchemeGroupVersion.WithResource(resource)) {
+		controllerStorage, err := controllerstore.NewStorage(restOptionsGetter)
+		if err != nil {
+			return kubernetesservice.RangeRegistries{}, genericapiserver.APIGroupInfo{}, err
+		}
+
 		storage[resource] = controllerStorage.Controller
 		storage[resource+"/status"] = controllerStorage.Status
 		if legacyscheme.Scheme.IsVersionRegistered(schema.GroupVersion{Group: "autoscaling", Version: "v1"}) {
 			storage[resource+"/scale"] = controllerStorage.Scale
 		}
 	}
 
+	// potentially override generic storage for service account (with pod support)
+	if resource := "serviceaccounts"; serviceAccountStorage != nil && apiResourceConfigSource.ResourceEnabled(corev1.SchemeGroupVersion.WithResource(resource)) {
+		// don't leak go routines
+		storage[resource].Destroy()
+		if storage[resource+"/token"] != nil {
+			storage[resource+"/token"].Destroy()
+		}
+
+		storage[resource] = serviceAccountStorage
+		if serviceAccountStorage.Token != nil {
+			storage[resource+"/token"] = serviceAccountStorage.Token
+		}
+	}
+
```

Comment:
this block is the only non-move bit in this PR. The generic func create a service account storage without pod support. Here we overwrite it.

Am open for other patterns, didn't find one that I like.

Reply:
I don't have a better one on my mind either...